hdf5 not installing with intel-mpi

I am trying to install hdf5 with intel mpi, but I get the following error:
```console
$ spack install hdf5@1.10.5^intel-mpi@2019.4.243
==> intel-mpi is already installed in /usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-9.1.0/intel-mpi-2019.4.243-u2ukgstakhmzbvzhx6at22bxq3jtzbub
==> zlib is already installed in /usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-9.1.0/zlib-1.2.11-wlsguaqpnjri6ufyhgrxdqw7p4otzsrf
==> Installing hdf5
==> Searching for binary cache of hdf5
==> Warning: No Spack mirrors are currently configured
==> No binary for hdf5 found: installing from source
==> Using cached archive: /usr/local/spack/var/spack/cache/hdf5/hdf5-1.10.5.tar.gz
==> Staging archive: /usr/local/spack/var/spack/stage/hdf5-1.10.5-ng6znwkpqz3bnrsgcoyqfsjbifjtexxj/hdf5-1.10.5.tar.gz
==> Created stage in /usr/local/spack/var/spack/stage/hdf5-1.10.5-ng6znwkpqz3bnrsgcoyqfsjbifjtexxj
==> No patches needed for hdf5
==> Building hdf5 [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> Error: ProcessError: Command exited with status 77:
    '/usr/local/spack/var/spack/stage/hdf5-1.10.5-ng6znwkpqz3bnrsgcoyqfsjbifjtexxj/spack-src/configure' '--prefix=/usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-9.1.0/hdf5-1.10.5-ng6znwkpqz3bnrsgcoyqfsjbifjtexxj' '--enable-unsupported' '--enable-symbols=yes' '--disable-threadsafe' '--disable-cxx' '--disable-hl' '--disable-fortran' '--without-szlib' '--enable-build-mode=production' '--enable-shared' 'CFLAGS=-fPIC' 'CXXFLAGS=-fPIC' 'FCFLAGS=-fPIC' '--enable-parallel' 'CC=/usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-9.1.0/intel-mpi-2019.4.243-u2ukgstakhmzbvzhx6at22bxq3jtzbub/compilers_and_libraries_2019.4.243/linux/mpi/intel64/bin/mpicc' '--with-zlib=/usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-9.1.0/zlib-1.2.11-wlsguaqpnjri6ufyhgrxdqw7p4otzsrf'

2 errors found in build log:
     30    checking for config ./config/site-specific/host-ssg.ucar.edu... no
     31    checking for config ./config/site-specific/host-ucar.edu... no
     32    checking for config ./config/site-specific/host-edu... no
     33    checking build mode... production
     34    checking for gcc... /usr/local/spack/opt/spack/linux-centos7-x86_64/
           gcc-9.1.0/intel-mpi-2019.4.243-u2ukgstakhmzbvzhx6at22bxq3jtzbub/comp
           ilers_and_libraries_2019.4.243/linux/mpi/intel64/bin/mpicc
     35    checking whether the C compiler works... no
  >> 36    configure: error: in `/usr/local/spack/var/spack/stage/hdf5-1.10.5-n
           g6znwkpqz3bnrsgcoyqfsjbifjtexxj/spack-src':
  >> 37    configure: error: C compiler cannot create executables
     38    See `config.log' for more details

See build log for details:
  /usr/local/spack/var/spack/stage/hdf5-1.10.5-ng6znwkpqz3bnrsgcoyqfsjbifjtexxj/spack-build-out.txt

$ cat /usr/local/spack/var/spack/stage/hdf5-1.10.5-ng6znwkpqz3bnrsgcoyqfsjbifjtexxj/spack-build-out.txt
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> [2019-08-20-20:10:17.374376] '/usr/local/spack/var/spack/stage/hdf5-1.10.5-ng6znwkpqz3bnrsgcoyqfsjbifjtexxj/spack-src/configure' '--prefix=/usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-9.1.0/hdf5-1.10.5-ng6znwkpqz3bnrsgcoyqfsjbifjtexxj' '--enable-unsupported' '--enable-symbols=yes' '--disable-threadsafe' '--disable-cxx' '--disable-hl' '--disable-fortran' '--without-szlib' '--enable-build-mode=production' '--enable-shared' 'CFLAGS=-fPIC' 'CXXFLAGS=-fPIC' 'FCFLAGS=-fPIC' '--enable-parallel' 'CC=/usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-9.1.0/intel-mpi-2019.4.243-u2ukgstakhmzbvzhx6at22bxq3jtzbub/compilers_and_libraries_2019.4.243/linux/mpi/intel64/bin/mpicc' '--with-zlib=/usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-9.1.0/zlib-1.2.11-wlsguaqpnjri6ufyhgrxdqw7p4otzsrf'
checking for a BSD-compatible install... /bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking shell variables initial values... done
checking if basename works... yes
checking if xargs works... yes
checking for cached host... none
checking for config x86_64-unknown-linux-gnu... no
checking for config x86_64-unknown-linux-gnu... no
checking for config unknown-linux-gnu... no
checking for config unknown-linux-gnu... no
checking for config x86_64-linux-gnu... no
checking for config x86_64-linux-gnu... no
checking for config x86_64-unknown... no
checking for config linux-gnu... found
compiler '/usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-9.1.0/intel-mpi-2019.4.243-u2ukgstakhmzbvzhx6at22bxq3jtzbub/compilers_and_libraries_2019.4.243/linux/mpi/intel64/bin/mpicc' is GNU gcc-9.1.0
compiler '/usr/local/spack/lib/spack/env/gcc/gfortran' is GNU gfortran-9.1.0
compiler '/usr/local/spack/lib/spack/env/gcc/g++' is GNU g++-9.1.0
checking for config ./config/site-specific/host-hl02.ssg.ucar.edu... no
checking for config ./config/site-specific/host-ssg.ucar.edu... no
checking for config ./config/site-specific/host-ucar.edu... no
checking for config ./config/site-specific/host-edu... no
checking build mode... production
checking for gcc... /usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-9.1.0/intel-mpi-2019.4.243-u2ukgstakhmzbvzhx6at22bxq3jtzbub/compilers_and_libraries_2019.4.243/linux/mpi/intel64/bin/mpicc
checking whether the C compiler works... no
configure: error: in `/usr/local/spack/var/spack/stage/hdf5-1.10.5-ng6znwkpqz3bnrsgcoyqfsjbifjtexxj/spack-src':
configure: error: C compiler cannot create executables
See `config.log' for more details
...
```
[config.log](https://github.com/spack/spack/files/3523080/config.log)

I am able to compile with my C compiler:
```console
$ cat test.c
#include<stdio.h>
int main(){
  printf("Hello World!\n");
  return 0;
}
$ gcc test.c
$ ./a.out
Hello World!
```

### Information on my system

```console
$ cat /etc/centos-release
CentOS Linux release 7.5.1804 (Core)
$ uname -r
3.10.0-693.el7.x86_64
$ cat /usr/local/spack/etc/spack/compilers.yaml
compilers:
- compiler:
    environment: {}
    extra_rpaths: []
    flags: {}
    modules: []
    operating_system: centos7
    paths:
      cc: /usr/bin/gcc
      cxx: /usr/bin/g++
      f77: /usr/bin/gfortran
      fc: /usr/bin/gfortran
    spec: gcc@4.8.5
    target: x86_64
- compiler:
    environment: {}
    extra_rpaths: []
    flags: {}
    modules: []
    operating_system: centos7
    paths:
      cc: /usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-9.1.0-mj7s6dgfnhgi2n42fyxgmitnuslcyol3/bin/gcc
      cxx: /usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-9.1.0-mj7s6dgfnhgi2n42fyxgmitnuslcyol3/bin/g++
      f77: /usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-9.1.0-mj7s6dgfnhgi2n42fyxgmitnuslcyol3/bin/gfortran
      fc: /usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-9.1.0-mj7s6dgfnhgi2n42fyxgmitnuslcyol3/bin/gfortran
    spec: gcc@9.1.0
    target: x86_64
- compiler:
    environment: {}
    extra_rpaths: []
    flags: {}
    modules: []
    operating_system: centos7
    paths:
      cc: /usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-5.5.0-iaqd4pqrvklcasb73bn3ugs6zx7wcghh/bin/gcc
      cxx: /usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-5.5.0-iaqd4pqrvklcasb73bn3ugs6zx7wcghh/bin/g++
      f77: /usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-5.5.0-iaqd4pqrvklcasb73bn3ugs6zx7wcghh/bin/gfortran
      fc: /usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-5.5.0-iaqd4pqrvklcasb73bn3ugs6zx7wcghh/bin/gfortran
    spec: gcc@5.5.0
    target: x86_64
$ spack config blame compilers
---                                           compilers:
/usr/local/spack/etc/spack/compilers.yaml:2   - compiler:
/usr/local/spack/etc/spack/compilers.yaml:3       environment: {}
/usr/local/spack/etc/spack/compilers.yaml:4       extra_rpaths: []
/usr/local/spack/etc/spack/compilers.yaml:5       flags: {}
/usr/local/spack/etc/spack/compilers.yaml:6       modules: []
/usr/local/spack/etc/spack/compilers.yaml:7       operating_system: centos7
/usr/local/spack/etc/spack/compilers.yaml:8       paths:
/usr/local/spack/etc/spack/compilers.yaml:9         cc: /usr/bin/gcc
/usr/local/spack/etc/spack/compilers.yaml:10        cxx: /usr/bin/g++
/usr/local/spack/etc/spack/compilers.yaml:11        f77: /usr/bin/gfortran
/usr/local/spack/etc/spack/compilers.yaml:12        fc: /usr/bin/gfortran
/usr/local/spack/etc/spack/compilers.yaml:13      spec: gcc@4.8.5
/usr/local/spack/etc/spack/compilers.yaml:14      target: x86_64
/usr/local/spack/etc/spack/compilers.yaml:15  - compiler:
/usr/local/spack/etc/spack/compilers.yaml:16      environment: {}
/usr/local/spack/etc/spack/compilers.yaml:17      extra_rpaths: []
/usr/local/spack/etc/spack/compilers.yaml:18      flags: {}
/usr/local/spack/etc/spack/compilers.yaml:19      modules: []
/usr/local/spack/etc/spack/compilers.yaml:20      operating_system: centos7
/usr/local/spack/etc/spack/compilers.yaml:21      paths:
/usr/local/spack/etc/spack/compilers.yaml:22        cc: /usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-9.1.0-mj7s6dgfnhgi2n42fyxgmitnuslcyol3/bin/gcc
/usr/local/spack/etc/spack/compilers.yaml:23        cxx: /usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-9.1.0-mj7s6dgfnhgi2n42fyxgmitnuslcyol3/bin/g++
/usr/local/spack/etc/spack/compilers.yaml:24        f77: /usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-9.1.0-mj7s6dgfnhgi2n42fyxgmitnuslcyol3/bin/gfortran
/usr/local/spack/etc/spack/compilers.yaml:25        fc: /usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-9.1.0-mj7s6dgfnhgi2n42fyxgmitnuslcyol3/bin/gfortran
/usr/local/spack/etc/spack/compilers.yaml:26      spec: gcc@9.1.0
/usr/local/spack/etc/spack/compilers.yaml:27      target: x86_64
/usr/local/spack/etc/spack/compilers.yaml:28  - compiler:
/usr/local/spack/etc/spack/compilers.yaml:29      environment: {}
/usr/local/spack/etc/spack/compilers.yaml:30      extra_rpaths: []
/usr/local/spack/etc/spack/compilers.yaml:31      flags: {}
/usr/local/spack/etc/spack/compilers.yaml:32      modules: []
/usr/local/spack/etc/spack/compilers.yaml:33      operating_system: centos7
/usr/local/spack/etc/spack/compilers.yaml:34      paths:
/usr/local/spack/etc/spack/compilers.yaml:35        cc: /usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-5.5.0-iaqd4pqrvklcasb73bn3ugs6zx7wcghh/bin/gcc
/usr/local/spack/etc/spack/compilers.yaml:36        cxx: /usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-5.5.0-iaqd4pqrvklcasb73bn3ugs6zx7wcghh/bin/g++
/usr/local/spack/etc/spack/compilers.yaml:37        f77: /usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-5.5.0-iaqd4pqrvklcasb73bn3ugs6zx7wcghh/bin/gfortran
/usr/local/spack/etc/spack/compilers.yaml:38        fc: /usr/local/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-5.5.0-iaqd4pqrvklcasb73bn3ugs6zx7wcghh/bin/gfortran
/usr/local/spack/etc/spack/compilers.yaml:39      spec: gcc@5.5.0
/usr/local/spack/etc/spack/compilers.yaml:40      target: x86_64

$ spack find
==> 99 installed packages
-- linux-centos7-x86_64 / gcc@4.8.5 -----------------------------
autoconf@2.69    gdbm@1.18.1  libsigsegv@2.11  mpfr@3.1.6     readline@7.0
automake@1.16.1  gmp@6.1.2    libtool@2.4.6    ncurses@6.1    zlib@1.2.11
gcc@5.5.0        isl@0.15     m4@1.4.18        perl@5.26.2
gcc@9.1.0        isl@0.19     mpc@1.1.0        pkgconf@1.6.1

-- linux-centos7-x86_64 / gcc@5.5.0 -----------------------------
autoconf@2.69     gettext@0.19.8.1     libxml2@2.9.9   readline@7.0
automake@1.16.1   help2man@1.47.8      m4@1.4.18       tar@1.31
bison@3.0.5       libiconv@1.15        mvapich2@2.3.1  texinfo@6.5
bzip2@1.0.8       libpciaccess@0.13.5  mvapich2@2.3.1  util-macros@1.19.1
diffutils@3.7     libsigsegv@2.11      ncurses@6.1     xz@5.2.4
findutils@4.6.0   libtool@2.4.6        perl@5.26.2     zlib@1.2.11
gdbm@1.18.1       libuuid@1.0.3        pkgconf@1.6.1
gettext@0.19.8.1  libxml2@2.9.9        psm@3.3

-- linux-centos7-x86_64 / gcc@9.1.0 -----------------------------
autoconf@2.69     hdf5@1.10.5           libxml2@2.9.9   python@2.7.16
automake@1.16.1   hwloc@1.11.11         lz4@1.9.0       python@3.6.1
bzip2@1.0.8       intel-mpi@2019.4.243  m4@1.4.18       readline@7.0
curl@7.63.0       json-c@0.13.1         mpich@3.3.1     slurm@18-08-0-1
diffutils@3.7     libbsd@0.9.1          munge@0.5.13    sqlite@3.28.0
emacs@26.1        libevent@2.1.8        nano@2.6.3      tar@1.31
expat@2.2.5       libffi@3.2.1          ncurses@6.1     texinfo@6.5
findutils@4.6.0   libgcrypt@1.8.4       numactl@2.0.12  tmux@2.8
gdbm@1.18.1       libgpg-error@1.36     openmpi@3.1.4   util-macros@1.19.1
gettext@0.19.8.1  libiconv@1.15         openssl@1.1.1b  vim@8.1.0338
glib@2.56.3       libpciaccess@0.13.5   pcre@8.42       xz@5.2.4
hdf5@1.10.5       libsigsegv@2.11       perl@5.26.2     zlib@1.2.11
hdf5@1.10.5       libtool@2.4.6         pkgconf@1.6.1

```

--- a/lib/spack/spack/build_systems/intel.py
+++ b/lib/spack/spack/build_systems/intel.py
@@ -1073,6 +1073,15 @@ class IntelPackage(PackageBase):
                 # which performs dizzyingly similar but necessarily different
                 # actions, and (b) function code leaves a bit more breathing
                 # room within the suffocating corset of flake8 line length.
+
+                # Intel MPI since 2019 depends on libfabric which is not in the
+                # lib directory but in a directory of its own which should be
+                # included in the rpath
+                if self.version >= ver('2019'):
+                    d = ancestor(self.component_lib_dir('mpi'))
+                    libfabrics_path = os.path.join(d, 'libfabric', 'lib')
+                    env.append_path('SPACK_COMPILER_EXTRA_RPATHS',
+                                    libfabrics_path)
             else:
                 raise InstallError('compilers_of_client arg required for MPI')
 

